FilmShader: Fix deterioration.

kaisavi opened the pull request:
No Related Issue

**Description**
Incorrect usage of the ``rand`` function causes the film effect to deteriorate with large time values. This is fixed by applying a modulus to the time value, restraining it to the expected [0,1]x[0,1] range of the ``rand`` function

https://github.com/mrdoob/three.js/blob/76e1fb171af400afebbfb851ef7d7297625c5f0a/src/renderers/shaders/ShaderChunk/common.glsl.js#L24

kaisavi committed:
Fix FilmShader Deterioration

Incorrect usage of the ``rand`` function causes the film effect to deteriorate with large time values. This is fixed by applying a modulus to the time value, restraining it to the expected [0,1]x[0,1] range of the ``rand`` function

https://github.com/mrdoob/three.js/blob/76e1fb171af400afebbfb851ef7d7297625c5f0a/src/renderers/shaders/ShaderChunk/common.glsl.js#L24

gkjohnson commented on examples/jsm/shaders/FilmShader.js:
```diff
@@ -38,7 +38,7 @@ const FilmShader = {
 			vec4 base = texture2D( tDiffuse, vUv );
 
-			float noise = rand( vUv + time );
+			float noise = rand( vUv + mod(time,1.0) );
```
> rand() coment:
> expects values in the range of [0,1]x[0,1]

This will still pass results in that are > 1. The modulus should wrap `vUv + time`.

gkjohnson reviewed (commented):
Can you provide screenshots showing what the "deterioration" looks like before and after this change.

kaisavi commented:
Noise function at 0 seconds
![0sec](https://github.com/mrdoob/three.js/assets/11150147/3daf4d3e-7fd1-491b-b0ec-7dfaf595c0da)

Noise function at 300 seconds
![300sec](https://github.com/mrdoob/three.js/assets/11150147/b76d5197-5d98-49bc-9755-33e031285c72)

You can view a demonstration of this effect on [Shadertoy](https://www.shadertoy.com/view/csVcDV)

kaisavi committed:
Use fract() & stricter constraint

Addresses https://github.com/mrdoob/three.js/pull/26908#discussion_r1349323634
the constraining function is now fract() instead of mod(n,1.0), and wraps vUv + time instead of only time

Mugen87 committed:
Update FilmShader.js

Clean up.
